tests/test_documents.py

```diff
@@ -161,3 +166,101 @@ async def test_add_document_ingest_error(client):
         client=client, collection=collection.name, document_id=document.id
     )
     assert len(chunks) == 0
+
+async def test_upload_document_content_invalid(client, doc_fixture):
+    with open(NEARLY_EMPTY_PATH, 'rb') as file:
+        response = await upload_document_content.asyncio_detailed(
+            client=client,
+            document_id=1_000_000,
+            body=BodyUploadDocumentContent(
+                content=File(
+                    payload=file,
+                    file_name="file-name-1",
+                    mime_type="application/pdf",
+                ),
+            ),
+        )
+    assert response.status_code == 404
```

Comment:
Why does invalid content lead to a 404? I wonder if we should start adding a comment on tests?

Or is this `upload_document_content` `invalid_document`?

Reply:
This gets into REST arcana which isn't my favorite, but the `document_id` is pulled out of the path, and in this case refers to a document that doesn't exist, so in this case we're trying to post to a request URL the service doesn't know about (ie 404). I'll rename the test though, it's probably better called "upload_unknown_document" or something.